core-default/src/main/java/me/aurium/beetle/defaults/task/GenericTasker.java

```diff
@@ -1,45 +1,17 @@
 package me.aurium.beetle.defaults.task;
 
-import me.aurium.beetle.api.task.TaskQueue;
-import me.aurium.beetle.api.task.Runner;
-import me.aurium.beetle.api.task.Tasker;
-
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
-public class GenericTasker implements Tasker {
-
-    private final TaskQueue taskQueue;
-    private final Runner runner;
-    private final ExecutorService executor;
-
-    public GenericTasker() {
-        this.executor = Executors.newFixedThreadPool(5000);
-        this.taskQueue = new CommonTaskQueue();
-        this.runner = new GenericRunner(taskQueue,executor);
-    }
+public class GenericTasker extends AbstractTasker{
 
     public GenericTasker(ExecutorService service) {
-        this.executor = service;
-        this.taskQueue = new CommonTaskQueue();
-        this.runner = new GenericRunner(taskQueue,executor);
-    }
-
-    @Override
-    public TaskQueue getQueue() {
-        return this.taskQueue;
-    }
-
-    @Override
-    public Runner getRunner() {
-        return this.runner;
+        super(service);
     }
 
     @Override
     public void launch() {
-        CompletableFuture.delayedExecutor(50L, TimeUnit.MICROSECONDS).execute(taskQueue::tick);
+        CompletableFuture.delayedExecutor(50L, TimeUnit.MICROSECONDS).execute(asQueue()::tick);
```

Comment:
What is the purpose of GenericTasker given this is all it does? You are running `tick` on `ForkJoinPool.commonPool()`. If you are trying to make a Tasker for a platform which does not have a main thread, then it does not even make sense to have a taskQueue

Reply:
The purpose of this is to make a tasker without a main thread. What would be an acceptable solution?

Reply:
How else am i supposed to resync to the main thread?

Reply:
If there's no main thread, there's no re-syncing going on, so there's no sense in having a taskQueue with sync tasks in it. If there's no main thread, you have to live with concurrent operations - there is nowhere to hide from the concurrency unless you make your own single-thread thread pool and move your own operations to it

Reply:
GREAT IDEA (maybe not)

Reply:
I'll maybe design the api for commons around the possibility of a single thread thread collection, that way if i decide to do it it can be done